### DISCUSSION TITLE
Firebase 360 / Analytics 25 Parameter Limit, 100 is allowed for 360

### DISCUSSION THREAD

**Discussion by GKMindera** [invertase/react-native-firebase#5676] [Q&A] [answered]
An error gets thrown by the React Native Firebase wrapper (v5) when more than 25 parameters are emitted to Firebase. We have the Firebase 360 version that allows more than 25 parameters to be logged. 

Is there a React Native Firebase version that allows 25 more than params without throwing an error? Or is there a recommended way of handling this?

**Comment by mikehardy** [comment, marked as answer]
Hi there! v5 is quite old, I would definitely upgrade. That said, I was just in the analytics code recently for #5630 and there is still a 25 parameter limit. I wasn't aware that was variable! Can you point me to any documentation about this Firebase 360 thing - I'm totally unaware of it :sweat_smile: - if more than 25 parameters are valid for certain cases I'm sure we can accommodate that

It will be in a version after current though, so work on getting to v12.7.4 in the meanwhile so you'll be able to ingest the change if/when it happens...

**Reply by GKMindera** [reply]
![image](https://user-images.githubusercontent.com/71133807/131762059-74c086f5-3a55-4aff-84e4-f0de6ceba5d4.png)

Thanks @mikehardy for your swift reply. The 360 upgrade is in beta for 360 customers only so there is no documentation unfortunately.  Please see email by Google upgrading our Firebase account and disclosing our limit gets increased to 100 parameters. It's my understanding you guys work directly with Google? Could this be something to confirm with them for further confidence?

**Reply by mikehardy** [reply]
Yeah - we collaborate certainly. I can confirm but the email is certainly sufficient. To be clear it was not so much that I doubted - I'm not surprised this exists - I was just hoping there was a documentation site where I could poke around and also (hopefully) find a way to detect dynamically what type of account or what type of limit was in force so that I could dynamically enforce the correct limit. With no insight like that I'll probably just make a method on the analytics object that allows you to set a new limit, then during init you may call it to bump it to 100 and off we'll go...

**Reply by GKMindera** [reply]
Thats great @mikehardy.

If you can kindly please let me know or point me to announcement board if you decide to go ahead with this change that would be great. That means I can raise a change with the team when this feature is available. Thanks very much!

**Reply by mikehardy** [reply]
I'll try to remember to post here, but an easy way is to "watch" the repo with a custom watch on releases only, then check the changelog when a release is posted

**Comment by GKMindera** [comment]
Yes sure that's what I was after thank you. I didn't mean to ask for a
personalised service!

**Comment by gmczoop** [comment]
Hi all, I was about to enquire about this also. Very grateful for this react native firebase project, but I just wanted to flag that this current 25 param limit is a blocker on my end also as I use the Firebase payload to trigger GA events via GTM. I'm glad to hear the limit is being raised to 100, this will resolve the issue on my end. Thanks!

**Comment by mikehardy** [comment]
I've got PR #5687 open for this and I should be able to merge and release it shortly. Here's the proposed API:

https://github.com/invertase/react-native-firebase/pull/5687/files#diff-f2aa40411bfb84c4832f3fb58d660ca98c44fc5108f49b536dc474072a87ecae

```typescript
    /**
     * Allows an override to the default 25 event parameter limit. This may be called
     * for users with upgraded analytics accounts to raise the parameter limit to their actual
     * account limit, e.g. 100
     *
     * @param limit
     */
    setEventParamLimit(limit: number): Promise<void>;
```

Does that seem like it will work well for everyone?

**Reply by gmczoop** [reply]
Works well for me - just tested on my local. Thank you, much appreciated!

**Reply by mikehardy** [reply]
Glad to hear it functions! After internal review we may go a slightly different route though, we may simply remove the restriction entirely and let the underlying SDKs simply accept them or not. I still need to test the success and failure cases in that scenario though to make sure they are still developer friendyl

**Reply by gmczoop** [reply]
Ah okay makes sense - thanks again.

**Reply by mikehardy** [reply]
Confirmed my last comment - next release will simply remove the limit and go to a silent-fail scenario but we'll add documentation to the method. In DebugView the failures are shown clearly with an orange bug icon for the event and a link to clear documentation explaining the issue, so the developer experience isn't bad.

Stated differently: there will be no code changes needed in the future to log >25 event parameters, and if you need it right this moment, you either use the patch set on the action (https://github.com/invertase/react-native-firebase/actions/runs/1198984863) or reach right in to node_modules/@react-native-firebase/analytics/lib/index.js right now and simply delete the >25 check (using patch-package so it's safe!). That will be the change we release, so you'll just be beating us to release
